VenafiPS/Public/Add-TppAdaptableHash.ps1

```diff
@@ -0,0 +1,175 @@
+function Add-TppAdaptableHash {
+    <#
+    .SYNOPSIS
+    Adds or updates the hash value for an adaptable script
+
+    .DESCRIPTION
+    TPP stores a base64 encoded hash of the file contents of an adaptable script in the Secret Store. This is referenced by
+    the Attribute 'PowerShell Script Hash Vault Id' on the DN of the adaptable script. This script retrieves the hash (if
+    present) from the Secret Store and compares it to the hash of the file in one of the scripts directories. It then adds 
+    a new or updated hash if required. When updating an existing hash, it removes the old one from the Secret Store.
+
+    .PARAMETER Path
+    Required. Path to the object to add or update the hash.
+
+    .PARAMETER Class
+    Used when setting policy attributes for an Adaptable App. The only valid parameter is 'Adaptable App'.
+
+    .PARAMETER Keyname
+    The name of the Secret Encryption Key (SEK) to used when encrypting this item. Default is "Software:Default"
```

Comment:
Just asked Jason and this is only needed if you have multiple DEKs and want to use a particular one. I'm going to remove it.

Reply:
The docs on this are not too clear, so I asked PM. If an HSM is in use for DEK the value would be [HSM Connector]:[Keyname], which will appear in the UI instead of Software:Default. This is to allow for a different value to be used.